### PR TITLE
Update permissions for kubectl.config.yaml

### PR DESCRIPTION
**Pull request by Chemaclass**
## Description

After upgrading to helm 3, I faced this issue and I found a potential solution for this too: https://github.com/helm/helm/issues/9115
```
WARNING: Kubernetes configuration file is group/world-readable 
```

### CODE REVIEW

File: src/_base/harness/config/pipeline.yml [hunk 1]
```diff
@@ -88,6 +88,9 @@ command('app deploy <environment>'):
     set -o pipefail
     cd "${ENVIRONMENT}"
     doctl -t "$DO_ACCESS_TOKEN" kubernetes cluster kubeconfig show "$CLUSTER" > kubectl.config.yaml
+    if helm version --short --client | grep '^v3' >/dev/null 2>&1; then
+      chmod go-r kubectl.config.yaml
+    fi
```
Comment (andytson) [inline src/_base/harness/config/pipeline.yml]:
```suggestion
    (umask 0200 && doctl -t "$DO_ACCESS_TOKEN" kubernetes cluster kubeconfig show "$CLUSTER" > kubectl.config.yaml)
```

I think its fine making it for helm v2 as well. This makes it so the file never got go+r

**Comment by andytson** [comment]
the reason we never saw this is because we started using ArgoCD (which we use helm template render) before the switch to helm v3, so this code was never later used with helm v3 except I think just by your project.

File: src/_base/harness/config/pipeline.yml (hunk 1 above)
Comment (Chemaclass) [reply src/_base/harness/config/pipeline.yml]: Yeah, I think helm2 would be ok with this as well. Thanks! 🙏

File: src/_base/harness/config/pipeline.yml [hunk 2]
```diff
@@ -87,7 +87,7 @@ command('app deploy <environment>'):
     #!bash(harness:/helm)|=
     set -o pipefail
     cd "${ENVIRONMENT}"
-    doctl -t "$DO_ACCESS_TOKEN" kubernetes cluster kubeconfig show "$CLUSTER" > kubectl.config.yaml
+    (umask 0200 && doctl -t "$DO_ACCESS_TOKEN" kubernetes cluster kubeconfig show "$CLUSTER" > kubectl.config.yaml)
```
Comment (andytson) [inline src/_base/harness/config/pipeline.yml]: silly me, wrong way around in my last suggestion
```suggestion
    (umask 0077 && doctl -t "$DO_ACCESS_TOKEN" kubernetes cluster kubeconfig show "$CLUSTER" > kubectl.config.yaml)
```

Comment (Chemaclass) [reply src/_base/harness/config/pipeline.yml]: Is this equivalent as the proposed solution from the issue: https://github.com/helm/helm/issues/9115#issuecomment-818832622 ?
> chmod 600 ~/.kube/config

Comment (andytson) [reply src/_base/harness/config/pipeline.yml]: yes, umask removes the bits that are allowed on a file on creation, so e.g. 0007 means if a file created with 666 permissions then it gets 660 permissions. You can see it easily with `(umask 0077 && touch testfile)` that it gets 600 permissions

File: src/_base/harness/config/pipeline.yml [hunk 3]
```diff
@@ -87,7 +87,7 @@ command('app deploy <environment>'):
     #!bash(harness:/helm)|=
     set -o pipefail
     cd "${ENVIRONMENT}"
-    doctl -t "$DO_ACCESS_TOKEN" kubernetes cluster kubeconfig show "$CLUSTER" > kubectl.config.yaml
+    (umask 0077 && doctl -t "$DO_ACCESS_TOKEN" kubernetes cluster kubeconfig show "$CLUSTER" > kubectl.config.yaml)
```
Comment (Chemaclass) [inline src/_base/harness/config/pipeline.yml:90]: For the [record](https://github.com/inviqa/harness-base-php/pull/710#discussion_r876336860): `umask` removes the bits that are allowed on a file on creation, so e.g. 0007 means if a file created with 666 permissions then it gets 660 permissions. You can see it easily with (umask 0077 && touch testfile) that it gets 600 permissions.

**Pull request merged**